components/single_image_view.jsx

```diff
@@ -102,36 +106,29 @@ export default class SingleImageView extends React.PureComponent {
         this.fileImage = node;
     }
 
-    computeImageDimension() {
+    computeImageHeight = () => {
         const {fileInfo} = this.props;
         const {viewPortWidth} = this.state;
 
-        let previewWidth = fileInfo.width;
         let previewHeight = fileInfo.height;
 
-        if (previewWidth > viewPortWidth) {
+        if (viewPortWidth && fileInfo.width > viewPortWidth) {
             const origRatio = fileInfo.height / fileInfo.width;
-            previewWidth = Math.floor(Math.min(PREVIEW_IMAGE_MAX_WIDTH, fileInfo.width, viewPortWidth));
-            previewHeight = Math.floor(previewWidth * origRatio);
+            previewHeight = Math.floor(fileInfo.width * origRatio);
         }
 
         if (previewHeight > PREVIEW_IMAGE_MAX_HEIGHT) {
-            const heightRatio = PREVIEW_IMAGE_MAX_HEIGHT / previewHeight;
             previewHeight = PREVIEW_IMAGE_MAX_HEIGHT;
-            previewWidth = Math.floor(previewWidth * heightRatio);
         }
 
-        return {previewWidth, previewHeight};
+        return previewHeight;
```

Comment:
I don't really think computing and setting the width was useful at all, it looks like the CSS was handling those cases fine based on my testing

Reply:
It's useful during initial load of image where image should scale up/down based on available browser window. Actually such feature was lost after we dealt with the edge case like (1) very tall but very thin, or (2) very wide but very short image.

Take an example of full AboutModal screenshot, on small window it didn't resize. Instead, it scaled down only based on max preview height then cropped the right part of the image.
![screen shot 2018-02-08 at 9 30 05 pm](https://user-images.githubusercontent.com/5334504/35975331-53808f8e-0d17-11e8-94af-9f7632a823c0.png)


@asaadmahmood has a separate fix for that here https://github.com/mattermost/mattermost-webapp/pull/743

Reply:
Ok cool, if #743 fixes that then let's take that as well